### PR TITLE
Multiple user backends

### PR DESCRIPTION
**Pull request by exekias**
This PR adds support for multiple user backends at conf level, that means, having the users split between several ADs.

Configuration is backwards compatible but allows to define and provision multiple servers

It works for sogo, panel, postfix and dovecot. For Openchange you need to deploy different pools for each samba

### CODE REVIEW

File: modules/zentyalcloud/templates/postfix/main.cf.erb [hunk 1]
```diff
@@ -51,10 +61,12 @@ smtpd_tls_mandatory_ciphers = medium
...
 tls_random_source = dev:/dev/urandom
 
-# Virtual Domains
+##
+## Virtual domains and alias
+##
 dovecot_destination_recipient_limit = 1
 virtual_transport = lmtp:inet:<%= @lmtp_host %>:24
-virtual_mailbox_domains = ldap:/etc/postfix/vdomains.cf
+virtual_mailbox_domains = <% @ldap_servers.each do |s| -%> ldap:/etc/postfix/vdomains-<%= s['realm'] %>.cf<% end %>
```
Comment (figarocorso) [inline modules/zentyalcloud/templates/postfix/main.cf.erb]: Nice!

File: envs/distributed/Vagrantfile [hunk 2]
```diff
@@ -52,14 +52,28 @@ Vagrant.configure("2") do |config|
...
+  end
+
+  # Samba 2
+  config.vm.define "dc2" do |dc1|
+    dc1.vm.hostname = "dc2.devel.zentyal.lan"
+    dc1.vm.network "private_network", ip: "192.168.56.15"
+    dc1.vm.provider "virtualbox" do |v|
+        v.memory = 512
+    end
+    dc1.vm.provision :puppet do |puppet|
```
Comment (figarocorso) [inline envs/distributed/Vagrantfile]: dc1.vm....? shouldn't it be dc2.vm....?
Not sure here as I usually build Vagrantfiles by trial-and-error.

Comment (exekias) [inline envs/distributed/Vagrantfile]: totally

**Comment by exekias** [comment]
Taking into account https://github.com/Zentyal/zinc-puppet/pull/51 changes, I will rework this before merging

**Pull request closed without merging**